File: back/api/DTOs/Address.cs
namespace api.DTOs;

public record CreateAddressDTO
{
  public string Name { get; init; }
  public string PhoneNumber { get; init; }
  public string Email { get; init; }
  public string StreetAddress { get; init; }
  public string City { get; init; }
  public string State { get; init; }
  public string Zip { get; init; }
}

public record UpdateAddressDTO
{
  public string? Name { get; init; }
  public string? PhoneNumber { get; init; }
  public string? Email { get; init; }
  public string StreetAddress { get; init; }
  public string? City { get; init; }
  public string? State { get; init; }
  public string? Zip { get; init; }
}

File: packages/database/src/models/users/Address.ts
import mongoose, { Schema } from "mongoose";
import { AddressDocument } from "shared2";
import { addressModelName } from "./modelNames";

export const AddressModel = mongoose.modelNames().includes(addressModelName)
  ? (mongoose.models[addressModelName] as mongoose.Model<AddressDocument, {}, {}, {}>)
  : mongoose.model<AddressDocument>(
      "game",
      new mongoose.Schema<AddressDocument>(
        {
          userId: { type: Schema.Types.ObjectId, required: true },
          name: { type: String, required: true },
          phoneNumber: { type: String },
          email: { type: String, required: true },
          streetAddress: { type: String },
          city: { type: String },
          state: { type: String },
          zip: { type: String },
        },
        { timestamps: true, versionKey: false }
      )
    );
